### ISSUE TITLE
Support NVIDIA GPUs

### ISSUE THREAD

**Issue by mavogel** [kreuzwerker/terraform-provider-docker#63]
<i>This issue was originally opened by @captn3m0 as https://github.com/hashicorp/terraform-provider-docker/issues/257. It was migrated here as a result of the community provider takeover from @kreuzwerker. The original body of the issue is below.</i><hr>
### Terraform Version
`v0.12.23`

### Affected Resource(s)
Please list the resources as a list, for example:
- `docker_container`

### Terraform Configuration Files
```hcl
resource docker_container a {
  name = ""
  gpus = "all"
}
```

### Debug Output

```
docker_container.a: : invalid or unknown key: gpus
```

### Expected Behavior
The `--gpus` flag should be supported.

### Actual Behavior
The `--gpus` flag is not supported.

### Steps to Reproduce
Try to run a container with NVIDIA Docker support, which is now natively available in Docker 19.03

### Important Factoids

- NVIDIA has a FAQ on the feature [here](https://github.com/NVIDIA/nvidia-docker/wiki)
- The docker Engine API implementation sends a `DeviceRequest` to the update/create container endpoints. See [this PR](https://github.com/moby/moby/pull/38828/files) for details.
    >* `POST /containers/create` now accepts `DeviceRequests` as part of `HostConfig`.
  Can be used to set Nvidia GPUs

**Comment by captn3m0** [comment]
Commenting to re-open.

**Comment by NeinsKnull** [comment]
We need that as well, your work here would be much appreciated!

**Comment by matifali** [comment]
Is there a gpu flag now?

**Comment by Junkern** [comment]
I do not have a nVidia GPU. Can anyone from this issue checkout the branch from [this PR](https://github.com/kreuzwerker/terraform-provider-docker/pull/405) and test the feature?

**Comment by Junkern** [comment]
Released with `v2.19.0`
